Documentation update (#2)
open-xcm-tools/xcm-js-tools: packages/xcm-util/README.md

## Installation

Before installing `xcm-util` package to your project, you need to setup Yarn 2, due to requirements of the `open-xcm-tools` packages.

### Yarn 2 (Berry) installation

First, set up a global Yarn, which will be used to create local instances:

```bash
npm install -g yarn
```

Then you need to locally change **Yarn** version for your project. In project folder:

```bash
yarn set version berry
```

After this actions you can use **Yarn** as usually (init project, add dependencies, etc.)

### `xcm-util` installation

```bash
yarn add @open-xcm-tools/xcm-util
```

## Usage

> For showing all functionality let's take a look to the examples of `Location` and `Asset` entities:

Reviewer: >For showing all functionality

"all functionality"? This package is rather diverse; I don't get what you mean here.

```typescript

> const location: Location = location(1n, [

Reviewer: ```suggestion
// A more concise way to write `{ parents: 1n, interior: { x3: [/* parachain, ... */] } }`
const location: Location = location(1n, [
```

>   {parachain: 2001n},

Reviewer: Let it be USDT :)

```suggestion
  {parachain: 1000n},
```

  {palletInstance: 50n},
  {generalIndex: 1984n},
]);

> const asset: Asset = asset(location(0n, 'here'), fungible(100n));

Reviewer: ```suggestion
// A more concise way to write `{ id: { parents: 0n, interior: 'here' }, fun: { fungible: 100n } }`
const asset: Asset = asset(location(0n, 'here'), fungible(100n));
```

```

### Convert XCM version

The examples above are `Location` and `Asset` instance of latest XCM versions.  The package `xcm-util` provides the opportunity to convert current XCM version of instances to specific, for example:

```typescript
const locationV3: LocationV3 = convertLocationVersion(3, location);
const assetV3: AssetV3 = convertAssetVersion(3, asset);
```

Also you can create `VersionedAsset` instance of XCM version 2, for example, and convert it to the latest XCM version, using:

```typescript
const versionedAssetV2: VersionedAsset = {
  V2: {
    concrete: {
      interior: 'here',
      parents: 0n,
    },
    fun: {
      fungible: 10n,
    },
  },
};

const asset: Asset = assetIntoCurrentVersion(versionedAssetV2);
```

### Sorting and deduplicating assets

> `xcm-util` defines a set of functions designed to sort and deduplicate asset collections based on their versioning and properties. The primary goal is to ensure that the assets are organized in a specific order and that any duplicates are removed, while also aggregating certain properties of the assets when duplicates are found.

Reviewer: I smell ChatGPT again :)

>while also aggregating certain properties of the assets when duplicates are found.

Do you mean the sum of fungibles of the same asset ID? And disregarding duplicate NFTs?

Please rephrase it in a more direct way.

> `xcm-util` package provides opportunity to create an array of assets without thinking about their logical order, just use the function `sortAndDeduplicateAssets` or `sortAndDeduplicateVersionedAssets`:

Reviewer: ```suggestion
The `sortAndDeduplicateAssets` and `sortAndDeduplicateVersionedAssets` functions provide the described functionality.
```

```typescript
const assets = [
  asset(location(0n, 'here'), fungible(100n)),
  asset('DOT', fungible(5n)),
  xcm.adjustedFungible('DOT', '2.5'),

  <AssetLookup>{
    id: 'DOT',
    fun: fungible(42n),
  },

  <Asset>{
    id: {parents: 0n, interior: 'here'},
    fun: {fungible: 88n},
  },

  asset('QTZ', fungible(77n)),
];

sortAndDeduplicateAssets(assets);
```

> **Note:** `sortAndDeduplicateAssets` and `sortAndDeduplicateVersionedAssets` working with array in-place.

### Sanitizing XCM entities

> `xcm-util` provides a comprehensive set of functions for sanitizing and validating various assets data structures in the context of cross-chain messaging (XCM). It ensures that the data adheres to specified constraints, such as bit size and byte length, thereby preventing potential errors and inconsistencies when processing assets across different networks. The module includes functions to sanitize assets, fungibility types, locations, junctions, and network identifiers, ensuring that all data conforms to the expected formats and standards.

Reviewer: ```suggestion
`xcm-util` provides a comprehensive set of functions for sanitizing and validating various assets data structures in the context of cross-chain messaging (XCM). It ensures that the data adheres to specified constraints, such as bit size and byte length, thereby preventing potential errors and inconsistencies. The module includes functions to sanitize assets, fungibility types, locations, junctions, and network identifiers, ensuring that all data conforms to the expected formats and standards.
```

> This functionality will validate and sanitize your array of assets:

Reviewer: ```suggestion
For instance, the `sanitizeAssets` function will validate and sanitize your array of assets:
```

```typescript
const assets = [
  asset(location(0n, 'here'), fungible(100n)),
  asset('DOT', fungible(5n)),
  xcm.adjustedFungible('DOT', '2.5'),

  <AssetLookup>{
    id: 'DOT',
    fun: fungible(42n),
  },

  <Asset>{
    id: {parents: 0n, interior: 'here'},
    fun: {fungible: 88n},
  },

  asset('QTZ', fungible(77n)),

> ];

Reviewer: It won't work. `sanitizeAssets` doesn't know about Registry.

Please make a couple of examples where this function spots errors and one example that illustrates ss58 conversion.
